cli/src/commands/branch.rs

```diff
@@ -298,6 +317,39 @@ fn cmd_branch_create(
     Ok(())
 }
 
+fn cmd_branch_rename(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &BranchRenameArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let target_commit =
+        workspace_command.resolve_single_rev(args.revision.as_deref().unwrap_or("@"), ui)?;
+    let repo = workspace_command.repo().as_ref();
+    let new_branch = &args.new;
+    let new_target = repo.view().get_local_branch(new_branch);
+    if !new_target.is_absent() {
+        return Err(user_error_with_hint(
+            format!("Branch name conflict: {new_branch}"),
+            "Target branch name already exists.",
+        ));
+    }
+
+    let old_branch = &args.old;
+    let mut tx = workspace_command.start_transaction(&format!(
+        "rename {} to {}; still point to commit {}",
+        make_branch_term(&[old_branch]),
+        make_branch_term(&[new_branch]),
+        target_commit.id().hex()
+    ));
```

Comment:
I just merged #2694 earlier today so the description now needs to be passed to `tx.finish()` below instead.

Reply:
Fixed. thanks.